### PR TITLE
[mlir][vector] Fix vector.broadcast lowering for scalable vectors

### PR DESCRIPTION
**Pull request by banach-space**
This patch makes sure that the following case is lowered correctly
("duplication"):
```
  func.func @broadcast_scalable_duplication(%arg0: vector<[32]xf32>) -> vector<1x[32]xf32> {
    %res = vector.broadcast %arg0 : vector<[32]xf32> to vector<1x[32]xf32>
    return %res : vector<1x[32]xf32>
  }
```

### CODE REVIEW

File: mlir/lib/Dialect/Vector/Transforms/LowerVectorBroadcast.cpp [hunk 1]
```diff
@@ -85,7 +85,8 @@ class BroadcastOpLowering : public OpRewritePattern<vector::BroadcastOp> {
     if (srcRank < dstRank) {
       // Duplication.
       VectorType resType =
-          VectorType::get(dstType.getShape().drop_front(), eltType);
+          VectorType::get(dstType.getShape().drop_front(), eltType,
+                          dstType.getScalableDims().drop_front());
```
Comment (c-rhodes) [inline mlir/lib/Dialect/Vector/Transforms/LowerVectorBroadcast.cpp:89]: I noticed @MacDue has been doing:
```
VectorType::Builder(dstType).dropDim(0));
```

in his patches, which seems a bit nicer?

**Review by c-rhodes — APPROVED** [review]
One small suggestion but otherwise LGTM, cheers

File: mlir/lib/Dialect/Vector/Transforms/LowerVectorBroadcast.cpp (hunk 1 above)
Comment (MacDue) [reply mlir/lib/Dialect/Vector/Transforms/LowerVectorBroadcast.cpp:89]: Yep, that'd be preferred here :+1:

Comment (banach-space) [reply mlir/lib/Dialect/Vector/Transforms/LowerVectorBroadcast.cpp]: Done, thanks!

**Review by MacDue — APPROVED** [review]

**Pull request merged**